larky/src/main/java/com/verygood/security/larky/modules/vgs/vault/DefaultVault.java

```diff
@@ -0,0 +1,184 @@
+                throw Starlark.errorf(String.format(
+                        "Storage '%s' not found in available storage list [persistent, volatile]", storage
+                ));
+            }
+
+            return storageConfig.get(storage);
+        }
+
+        throw Starlark.errorf(String.format(
+                "Storage of type %s is not supported in DefaultVault, expecting String",
+                storage.getClass().getName()
+        ));
+    }
+
+    private AliasGenerator getTokenizer(Object format) throws EvalException {
+
+        if (format instanceof NoneType) {
+            return formatTokenizer.get("UUID"); // default tokenizer
+        } else if (format instanceof String) {
+            if (formatTokenizer.containsKey(format)) {
+                return formatTokenizer.get(format);
+            } else {
+                throw Starlark.errorf(String.format(
+                        "Format '%s' not found", format
+                ));
+            }
+        }
+
+        throw Starlark.errorf(String.format(
+                "Format of type %s is not supported in DefaultVault, expecting String",
+                format.getClass().getName()
+        ));
+    }
+
+    private class NumberLengthPreserving extends ValidatingAliasGenerator {
+
+        private final Pattern cardPattern = Pattern.compile("\\d{3,16}");
+
+        @Override
+        protected String internalTokenize(String value) {
+            return RandomStringUtils.randomNumeric(value.length());
+        }
+
+        @Override
+        protected boolean isValid(String value) {
+            return cardPattern.matcher(value).find();
+        }
+
+        @Override
+        protected AliasGenerator fallbackAliasGenerator() {
+            return new RawAliasGenerator();
+        }
+    }
+
+    private abstract class ValidatingAliasGenerator implements AliasGenerator {
+
+        @Override
+        public String tokenize(String value) throws EvalException {
+            if (!isValid(value)) {
+                return fallbackAliasGenerator().tokenize(value);
+            }
+            return internalTokenize(value);
+        }
+
+        protected abstract boolean isValid(String value);
+
+        protected abstract String internalTokenize(String Value);
+
+        protected abstract AliasGenerator fallbackAliasGenerator();
+
+    }
+
+    private class UUIDAliasGenerator extends RawAliasGenerator {
+        @Override
+        public String tokenize(String value) throws EvalException {
+            return String.format("tok_%s", super.tokenize(value)).substring(0, 30);
+        }
+    }
+
+    private class RawAliasGenerator implements AliasGenerator {
+        @Override
+        public String tokenize(String value) throws EvalException {
+            return new String(Base64.getEncoder().encode(UUID.randomUUID().toString().getBytes()));
+        }
+    }
+
+    private class NoopAliasGenerator implements AliasGenerator {
```

Comment:
There would be a ton more of generator, probably would be better to extrac them in separate class to not bloat this one. Let's have a separate `default` package for all of default vault stuff